src/components/FilmItem/index.js

```diff
@@ -1,16 +1,60 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
 
 import "./FilmItem.sass"
 
 import filmCardPlaceholder from './../../assets/images/placeholder_320x455_ffffff_cccccc.jpg'
 
-const FilmItem = ({ title, genre, release_date, thumbnail }) => {
+const FilmItem = ({ title, genre, release_date, thumbnail, onClickEdit, onClickDelete }) => {
     const checkFilmCardThumbnail = thumbnail ?
         thumbnail : filmCardPlaceholder;
+
+    const [showMenu, setShowMenu] = useState(false)
+    const menuRef = useRef(null)
+    const handleClickOutside = (event) => {
+        if (menuRef.current && !menuRef.current.contains(event.target)) {
+            setShowMenu(false)
+        }
+    }
+
+    useEffect(() => {
+        document.addEventListener('click', handleClickOutside, true)
+        return () => {
+            document.removeEventListener('click', handleClickOutside, true)
+        }
+    })
+
+    const clickToEditBtn = () => {
+        setShowMenu(false)
+        onClickEdit()
+    }
+    const clickToDeleteBtn = () => {
+        setShowMenu(false);
+        onClickDelete()
+    }
+
     return (
         <div className="film-card">
             <div className="film-card__in">
+                <div
+                    ref={menuRef}
+                    className={`film-card__menu ${showMenu ? 'film-card__menu--open' : null}`}>
+                    <button onClick={() => setShowMenu(true)}
+                        className="film-card__menu-toggle"
+                        aria-label="Film menu"></button>
+                    <div className="film-card__actions">
+                        <button
+                            onClick={() => setShowMenu(false)}
+                            className="film-card__actions-toggle"
+                            aria-label="Close"></button>
+                        <button onClick={clickToEditBtn} className="film-card__actions-item">
+                            Edit
+                        </button>
+                        <button onClick={clickToDeleteBtn} className="film-card__actions-item">
+                            Delete
+                        </button>
+                    </div>
+                </div>

```

Comment:
You could create a separate component (`CardMenu`) and use optional rendering:
```jsx
showMenu && <CardMenu />
```
But you could do it even without extracting logic into a separate component. It's up to you.
The main benefit of this approach is avoiding the creation of unnecessary DOM.

Reply:
If we are talking about selected nodes (`.film-card__menu` and `.film-card__menu-toggle`), then these nodes always need to render. These nodes visible by hover on `.film-card`.

I can create a separate component for nodes (`.card__actions` and `.card__actions-toggle`), but I think it overkill. 

However, extracting logic into a separate components is good idea. I implemented this way and made new component more flexible.

I didn't use condition render in new menu component, because if I used that, then I can't used simple and native CSS animation with transition. 

Maybe do you can advice some way to animate components when using condition rendering? I found way to animate with using custom hooks for change classes on rendering. It's looks like hard way for simple action.

Reply:
@aliskar, I updated modal component as example with using condition rendering and react hooks to solve animation problem